### PR TITLE
Fix for onMenuClick firing on every keyDown

### PR DESCRIPTION
**Pull request by Markionium**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Bryan pointed out to me that i introduced a bug during a previous fix. The `onMenuClick` for a button should only be fired in the `onKeyDown` handler when a key that should open the menu is pressed. 
In the current version it's fired on each `keyPress`

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -570,14 +570,14 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       this.props.onKeyDown(ev);
     }
 
-    const { onMenuClick } = this.props;
-    if (onMenuClick) {
-      onMenuClick(ev, this);
-    }
-
     if (!ev.defaultPrevented &&
       this.props.menuTriggerKeyCode !== null &&
```
Comment (manishgarg1) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: >this.props.menuTriggerKeyCode !== null [](start = 6, length = 38)

The "!== null" part is not required

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -570,14 +570,14 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       this.props.onKeyDown(ev);
     }
 
-    const { onMenuClick } = this.props;
-    if (onMenuClick) {
-      onMenuClick(ev, this);
-    }
-
     if (!ev.defaultPrevented &&
```
Comment (manishgarg1) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: >defaultPrevented [](start = 12, length = 16)

I don't see any change. The old and new code look same to me

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx (hunk 1 above)
Comment (Markionium) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Can remove it :)

**Comment by Markionium** [comment]
@manishgarg1 Lets hold off merging this one until Bryan confirms this is what fixes his issue? :)

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 3]
```diff
@@ -570,14 +570,14 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
...
-      this.props.menuTriggerKeyCode !== null &&
+      this.props.menuTriggerKeyCode &&
```
Comment (brwatt) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: It doesn't seem like you want to check `this.props.menuTriggerKeyCode` here? It is already being handled in `this._isValidMenuOpenKey`.

**Pull request merged**